src/main/java/picocli/CommandLine.java

```diff
@@ -13341,18 +13334,17 @@ private static String optionDescription(String prefix, ArgSpec argSpec, int opti
     private static String createMissingParameterMessage(ArgSpec argSpec, Range arity, List<PositionalParamSpec> missingList, Stack<String> args, int available) {
         if (arity.min == 1) {
             if (argSpec.isOption()) {
-                String msg = "Missing required parameter for " + optionDescription("", argSpec, 0);
-                return msg;
+                return "Missing required parameter for " + optionDescription("", argSpec, 0);
             }
             String sep = "";
-            String names = ": ";
-            String indices = "";
             String infix = " at index ";
+            StringBuilder names = new StringBuilder(": ");
+            StringBuilder indices = new StringBuilder();
             int count = 0;
             for (PositionalParamSpec missing : missingList) {
                 if (missing.arity().min > 0) {
-                    names += sep + "'" + missing.paramLabel() + "'";
-                    indices += sep + missing.index();
+                    names.append(sep).append('\'').append(missing.paramLabel()).append('\'');
+                    indices.append(sep).append(missing.index());
```

Comment:
Code style tools often suggest replacing `str1 + str2` with `StringBuilder.append`, often because this gave some performance benefit in older versions of Java.  However, as [this article](https://javapapers.com/java/java-string-vs-stringbuilder-vs-stringbuffer-concatenation-performance-micro-benchmark/) shows with some benchmarks, String concatenation with `+` is about as performant as `StringBuilder.append`, especially in short loops.

What is more important, I often find code that uses the `+` operator for String concatenation more readable than code that uses StringBuilder.append. Performance is not really an issue since the number of iterations will be very low.

Let's emphasize readability and keep the `+` style String concatenation.